### ISSUE TITLE
Elements Can Bypass Required or Other Validation

### ISSUE THREAD

**Issue by mwrouse** [surveyjs/survey-library#3062]
### Are you requesting a feature, reporting a bug or asking a question?
Reporting a Bug which also maybe a feature.

### What is the current behavior?
Given you have a checkbox group, radio select, or drop down field you can set a value such that it will pass validation. 
This validation can be a minimum count (for checkbox), or just marking it as required.

If you have a checkbox group that has a minimum count of 2, you can pass that validation by setting the value of 2 empty strings: `["", ""]`

If you have a radio button group or a dropdown marked as required you can pass validation by setting `['']`.

If you have a textfield it can get fooled into passing required validation by also setting it's value to an array `['']`

### What is the expected behavior?
I expect the validation of a checkbox, radio group, or dropdown to verify that the values given as input are valid values since the choices are specified. 

If the values are not valid then they should be ignored.

I would also expect all fields to do some kind of input checking (i.e., text field is smart enough to not be fooled by an empty string, but is fooled by an array)

The textbox seems less important to me than the checkbox because the output type of a checkbox element is an array, so an array input for a textbox shouldn't normally be expected.

### How would you reproduce the current behavior (if this is a bug)?
By setting the values specified above on a survey.

#### Provide the test code and the tested page URL (if applicable)

Tested page URL: https://jsfiddle.net/mwrouse/xhk4now2/20/

That page has 2 surveys on the same page, the first one is without setting any values, so it will work as expected. The second is setting the values that can trick the validation.

**Test code**

```javascript
var json = {
 "completedHtml": "<p><h4>Thank you for sharing this information with us.</h4></p><p>Your name is: <b>{name}</b></p><p>Your email is: <b>{email}</b></p><p>This information is not in the survey data result:<b> {tempvar}</b></p>",
 "pages": [
  {
   "name": "page1",
   "elements": [
    {
     "type": "checkbox",
     "name": "question1",
     "validators": [
      {
       "type": "answercount",
       "minCount": 2
      }
     ],
     "choices": [
      "item1",
      "item2",
      "item3"
     ]
    },
    {
     "type": "radiogroup",
     "name": "question2",
     "isRequired": true,
     "choices": [
      "item1",
      "item2",
      "item3"
     ]
    },
    {
     "type": "dropdown",
     "name": "question3",
     "isRequired": true,
     "choices": [
      "item1",
      "item2",
      "item3"
     ]
    },
     {
     "type": "text",
     "name": "question4",
     "isRequired": true
    }
   ],
   "title": "Survey Example"
  }
 ],
};

var survey = new Survey.Model(json);
survey.setValue('question1', ["", ""]);
survey.setValue('question2', ['']);
survey.setValue('question3', ['']);
survey.setValue('question4', ['']);
```

### Specify Your

* browser: Chrome
* browser version: Latest
* surveyjs platform (angular or react or jquery or knockout or vue): knockout
* surveyjs version: 1.8.55

**Comment by andrewtelnov** [comment]
@mwrouse If you are not sure about yor data that you assign in code, please use the [clearIncorrectValues()](https://surveyjs.io/Documentation/Library?id=surveymodel#clearIncorrectValues) function. 
Add this line after setting values into survey.
`survey2.clearIncorrectValues(true);`.

Thank you,
andrew

**Comment by mwrouse** [comment]
Nice. thanks. But adding it to my example it only solves the problem with the radio group, and the dropdown. 

The checkbox and the textfield questions still pass their validations.

**Comment by andrewtelnov** [comment]
@mwrouse The issue with checkbox I have fixed by the commit above. It doesn't work correctly with empty string, undefined and null. You can check the updated unit test.
Regarding other question types, you are setting an array. Questions dropdown/radiogroup/text box  don't expect to have an array as a value. Change it to empty string and it will work.

PS: We will include the fix into [v1.8.56](https://surveyjs.io/WhatsNew#v1.8.56).

Thank you,
Andrew
